### PR TITLE
fix(yarn2): only set immutable cache if cache available 

### PR DESCRIPTION
**Pull request by mlavina**
I ran into an issue where I didn't have cache committed into source because I was using node-linker and didn't think i needed that. 

I added this first commit to warn about that if someone else runs into that issue

But i was going to add a second commit to disable immutable cache if you are using node-linker

What are your thoughts @danielleadams @lillianzhang331 ?

Should we disable immutable-cache for all node-linkers? Should there be a different opt-out param similar to yarn_production? Should there be an opt-out at all?

### CODE REVIEW

File: lib/dependencies.sh [hunk 1]
```diff
@@ -109,6 +109,11 @@ yarn_2_install() {
 
   echo "Running 'yarn install' with yarn.lock"
   cd "$build_dir" || return
+
+  if ! has_yarn_cache "$build_dir"; then
+    echo "Yarn cache is required. See https://devcenter.heroku.com/articles/migrating-to-yarn-2#make-changes-to-source-code for additional information"
```
Comment (danielleadams) [inline lib/dependencies.sh:114]: What do you think about changing this to:
```bash

if ! has_yarn_cache "$build_dir"; then
  monitor "yarn-2-install-with-node-modules" yarn install 2>&1
else
 ...
```

Comment (mlavina) [reply lib/dependencies.sh:114]: that's what i thought of doing originally but i didn't want to be presumption with asking your team first haha.

**Comment by mlavina** [comment]
@danielleadams any ideas why the hatchet stuff is failing. I tested locally with make heroku-20. I can try a full make test but that's going to take a bit on my laptop

**Comment by mlavina** [comment]
QQ @lillianzhang331 will #833 solve the hatchet issue here. I am sorry, but I don't really understand what the hatchet test is and why it's failing in my PR

**Comment by lillianzhang331** [comment]
Hi @mlavina, #833 was opened to test existing behavior in the buildpack, so I don't think it conflicts with this PR. It wouldn't hurt to pull in those tests though!

**Comment by mlavina** [comment]
@lillianzhang331 I ran `make test` fully locally and it works. I don't understand why CI test=Hatchet is failing and everything is passing. The error on the CI seems to be
```
fetch': key not found: "HEROKU_API_USER" (KeyError)
```

Do I have to register my username somewhere?

**Comment by danielleadams** [comment]
@mlavina Hatchet tests are only supposed to run when members of Heroku team open pull requests (and when we create tests to check release acceptance). cc @lillianzhang331 We may want to check why the Hatchet tests are running when a contributors open PRs

**Comment by danielleadams** [comment]
(Not sure if it was always like that, but `HEROKU_API_USER` would refer to Heroku members and we have done some work on our CI recently, so it's likely we (ie. I) may have broken the testing workflow for contributors)

**Comment by danielleadams** [comment]
@mlavina do you mind rebasing this? I am back from vacation, so I can work on getting this in this week.

**Comment by mlavina** [comment]
@danielleadams rebased

**Comment by danielleadams** [comment]
@mlavina Going to fix these tests and then we can get it merged. Thanks again for the work!

**Review by danielleadams — APPROVED** [review]

**Comment by danielleadams** [comment]
There's a configuration the CI isn't picking up. I'm going to try closing the pull request and reopening to see if it picks it up.

**Comment by danielleadams** [comment]
@mlavina For some reason the Travis build is not picking up the environment variables - I think this is because the pull request was created before the variables were added. Do you mind creating a new pull request? Sorry for the additional work. No need to write an explanation - you can just link this PR because we'll want to reference the discussion here anyways. Thanks!

**Comment by mlavina** [comment]
sorry this is being so confusing i opened up #836

**Pull request closed without merging**